### PR TITLE
Prometheus: remove /series endpoint calls in query builder label names and values for supported clients

### LINKED ISSUE #58009: Prometheus: label filters - use /label match param
While we added support to use the Prometheus label match parameter in https://github.com/grafana/grafana/pull/57553 for the list of metrics, this didn't update the label filters endpoint, which is still calling the /series endpoint.

What I have noticed is first call (under Metric) is making a `resources/api/v1/label/__name__/values` call. But very next `Label Filters` calls are again `/resources/api/v1/series` 
Can we have `label` calls for Label filters as well?
image

_Originally posted by @adigov in https://github.com/grafana/grafana/issues/57553#issuecomment-1298899555_

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
For customers with lots of labels for a metric, the current /series call to get all of the label names and values for a selected metric (or for all metrics) can run quite slowly compared to the labels/ and label/ api endpoint. This proposed implementation would split out that one query into two, using the label name selected to further filter the values.

**Why do we need this feature?**
To improve performance of the UI.

**Who is this feature for?**
For customers with high cardinality databases and lots of metric labels/values.

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana/issues/58009

**Special notes for your reviewer**:
WIP still

### CODE REVIEW

**Comment by gtk-grafana** [comment]
So the initial implementation here "works", except that it still returns label values that don't have metrics, unlike the current implementation. This is a much more frustrating UX, and will need to be fixed before this implementation can proceed.

This PR:
image

9.2.3 
image

File: public/app/plugins/datasource/prometheus/language_provider.ts [hunk 1]
```diff
@@ -529,6 +545,47 @@ export default class PromQlLanguageProvider extends LanguageProvider {
       const { values } = processLabels(data, withName);
       value = values;
       this.labelsCache.set(cacheKey, value);
+      console.log('fetchSeriesLabels CACHE MISS');
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/language_provider.ts:548]: probably time to delete these

File: public/app/plugins/datasource/prometheus/language_provider.ts [hunk 2]
```diff
@@ -471,6 +471,10 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     }
   }
 
+  /**
+   * @todo cache
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/language_provider.ts:475]: This wasn't currently cached, do we want to attempt to cache these values like we do with the series result queries?

File: public/app/plugins/datasource/prometheus/language_provider.ts [hunk 3]
```diff
@@ -497,6 +501,18 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     return [];
   }
 
+  fetchSeriesValues = async (name: string, match?: string): Promise<string[]> => {
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/language_provider.ts:504]: Comment these and maybe spend some time thinking of better names?

File: public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ export interface Props {
   labelsFilters: QueryBuilderLabelFilter[];
 }
 
-const MAX_NUMBER_OF_RESULTS = 1000;
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx:25]: why do you have to refactor stuff that's outside the scope of this PR? smh, but it is a nicer name

Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx:25]: Oh yeah, we're exporting it now, so this is in scope, ignore me!

**Review by bohandley — COMMENTED** [review]
Looks mostly good! but when a datasource uses the `fetchSeriesLabelsMatch` with a query variable, this does not get interpolated in the label value input.

For example, with `fetchSeriesLabels` the label name 'instance' is interpolated and pass to the label value input which displays 'localhost:9095'.
Screen Shot 2022-11-04 at 10 44 19 AM

but with the `fetchSeriesLabelsMatch` we get an error as the template variable, $label_names is passed directly to  the label value input.
Screen Shot 2022-11-04 at 10 45 33 AM

**Comment by gtk-grafana** [comment]
@bohandley [This](https://github.com/grafana/grafana/pull/58087/commits/57118eb745a474a9efb8c591c22ea02bb2956c45) might explain the problem you were seeing! 🤦

**Review by bohandley — APPROVED** [review]
Tested it out with the labels values interpolation fix and LGTM!

**Review by itsmylife — APPROVED** [review]
LGTM 🚀

**Pull request merged**